sugar.go

```diff
@@ -213,6 +213,17 @@ func (p *Page) Screenshot(toFile ...string) []byte {
 	return bin
 }
 
+// FullScreenshot gets the full height of the page and returns the binary of the image.
+// Uses an internal JavaScript function to get the client height.
+func (p *Page) FullScreenshot(toFile ...string) []byte {
+	p.WaitLoad()
+	res, err := proto.PageGetLayoutMetrics{}.Call(p)
+	kit.E(err)
+	defer p.Viewport(int64(res.ContentSize.Width), int64(res.ContentSize.Height), 1, false)
```

Comment:
`defer` is used to recover the change not set the change.

Code should be something like:

```go
p.Viewport(...) // this method should set the private variable `page.viewport`
defer p.Viewport(p.viewport)
```

Reply:
I know it's boring to fix all those details, but I'm sure if you can make PR pass you will be much better at golang 😃

Reply:
Oh mate, I'm more than happy to go with the nitty gritty! 😄